src/twisted/names/common.py

```diff
@@ -67,8 +67,8 @@ def query(self, query, timeout=None):
         try:
             method = self.typeToMethod[query.type]
         except KeyError:
-            return defer.fail(failure.Failure(NotImplementedError(
```

Comment:
I added some logging; thanks for double-checking all the `ResolverBase` subclasses!